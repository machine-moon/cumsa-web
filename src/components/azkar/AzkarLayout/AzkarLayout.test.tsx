import React from "react";
import { render } from "@testing-library/react";

const MockAzkarLayout = () => (
  <div data-testid="azkar-layout">
    <div data-testid="category-list">Categories</div>
    <div data-testid="azkar-display">Azkar Display</div>
  </div>
);

describe("AzkarLayout", () => {
  it("renders correctly as snapshot", () => {
    const { container } = render(<MockAzkarLayout />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with mock structure", () => {
    const { container } = render(<MockAzkarLayout />);
    expect(container.querySelector('[data-testid="azkar-layout"]')).toBeInTheDocument();
    expect(container.querySelector('[data-testid="category-list"]')).toBeInTheDocument();
    expect(container.querySelector('[data-testid="azkar-display"]')).toBeInTheDocument();
  });
});
