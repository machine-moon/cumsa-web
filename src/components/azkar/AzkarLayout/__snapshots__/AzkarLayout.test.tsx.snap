// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AzkarLayout renders correctly as snapshot 1`] = `
<div
  data-testid="azkar-layout"
>
  <div
    data-testid="category-list"
  >
    Categories
  </div>
  <div
    data-testid="azkar-display"
  >
    Azkar Display
  </div>
</div>
`;
